llvm_util/llvm2alive.cpp

```diff
@@ -270,13 +270,30 @@ class llvm2alive_ : public llvm::InstVisitor<llvm2alive_, unique_ptr<Instr>> {
                                     !known);
     unique_ptr<Instr> ret_val;
     auto argI = fn->arg_begin(), argE = fn->arg_end();
+    auto alist = i.getAttributes();
+    uint64_t argidx = 0;
+
     for (auto &arg : args) {
       ParamAttrs attr;
+
+      if (uint64_t derefb = alist.getParamDereferenceableBytes(argidx)) {
+        attr.set(ParamAttrs::Dereferenceable);
+        attr.derefBytes = derefb;
+      }
+
       if (argI != argE) {
         // Check whether arg itr finished early because it was var arg
         if (argI->hasByValAttr())
           attr.set(ParamAttrs::ByVal);
-        else if (argI->hasReturnedAttr()) {
+
+        uint64_t derefb = argI->getType()->isPointerTy() ?
```

Comment:
can this not be a pointer??

Reply:
getDereferenceableBytes() fails if argI was not a pointer.